scipy/stats/tests/test_stats.py

```diff
@@ -3750,6 +3750,109 @@ def test_ttest_ind_permutations_many_dims(self):
         assert_equal(res.pvalue[i, :, j, k, :],
                      res2.pvalue)
 
+    def test_ttest_ind_exact_alternative(self):
+        np.random.seed(0)
+        N = 3
+        a = np.random.rand(2, N, 2)
+        b = np.random.rand(2, N, 2)
+
+        options_p = {'axis': 1, 'permutations': 1000}
+
+        options_p.update(alternative="greater")
```

Comment:
I like all these tests, they look like good candidates for `@pytest.mark.parametrize` though, is there a reason not to put them that way?

Reply:
I like parametrize as much as the next, really, but they're actually not great candidates. I'm not actually performing the same test on a lot of different combinations of parameters, for instance. I'm calculating a bunch of stuff - then testing the relationships between those calculations. 

I could share some code between `test_ttest_ind_exact_alternative`, `test_ttest_ind_randperm_alternative`, and `test_ttest_ind_randperm_alternative`, but there are some small tweaks between them so if the code were all shared, some of the calculations would not actually be needed/wanted every time. I thought about it, but decided against it.

Reply:
yeah I think that's a fair point here, there's a bunch of stuff going on in the test.